### ISSUE TITLE
S3 delete object should always return success even if not exists

### ISSUE THREAD

**Issue by guymguym** [noobaa/noobaa-core#7353]
### Environment info
- NooBaa Version: **5.12**
- Platform: **NA**

### Actual behavior
1. Delete object that doesn't exist returns NoSuchKey (404).

### Expected behavior
1. Should return success (204).
2. See https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html.

### Steps to reproduce
1. Delete a non existing object.

**Comment by guymguym** [comment]
@nimrod-becker @romayalon FYI.
@shirady This seems to be closely related to #7352.

**Comment by guymguym** [comment]
CC @cxhong

**Comment by romayalon** [comment]
@guymguym 
I fixed it for nsfs during the xattr folders PR - see delete_single_object() in namespace_fs.js
https://github.com/noobaa/noobaa-core/pull/7020/files

**Comment by guymguym** [comment]
@romayalon Thanks. We still need to resolve this for any namespace. Note that there are more calls outside that try-unlink-catch in delete_single_object, which might also fail, so I am thinking if we better off catch in the s3_delete_object handler level. This is what I did when creating this quick fix branch - https://github.com/guymguym/noobaa-core/blob/guy-5.12-delete-object-fix/src/endpoint/s3/ops/s3_delete_object.js

**Comment by romayalon** [comment]
@guymguym sure np we can go with fixing it anyway as you did, 
which means we need to convert errors to success on delete_multiple_objects as well.
While fixing it I checked the behavior of some different namespaces like namespace_nb and saw that we return success on the deletion of a non-existing object, Namespace S3 should return success etc, so I fixed it for nsfs on the namespace level as well.

**Comment by guymguym** [comment]
@romayalon Thinking about this some more, I also prefer fixing this inside each namespace level, because it's not that we really want to have a large try-catch around the whole operation and accidentally reply ok without really checking the object (for example if we fail early on checking for another file before the actual unlink). So your fix is a better direction going forward.

But we also need to make sure that all other namespaces will understand this delete object "missing-is-ok" semantics (namespace_blob / gcp / merge / cache / ....)

Regarding delete_multiple_objects - you're right, it's also using the same semantics, so we need to report it in the response list as deleted and not as error item. This is the aws docs:
> For each key, Amazon S3 performs a delete action and returns the result of that delete, success, or failure, in the response. Note that if the object specified in the request is not found, Amazon S3 returns the result as deleted.

**Comment by romayalon** [comment]
@guymguym Oh, I added fixes for both levels in https://github.com/noobaa/noobaa-core/pull/7356
so should I remove the outside "non-existing object" error to success conversion?

**Comment by rkomandu** [comment]
@romayalon , does this issue needs to be closed ? it looks like automatically closed

**Comment by rkomandu** [comment]
please close it if appropriate

**Comment by romayalon** [comment]
@rkomandu for NSFS it's fixed, but maybe not for other namespaces, some research work here needs to be done.

**Comment by rkomandu** [comment]
@romayalon has this fixed in the NC version ?

**Comment by romayalon** [comment]
yes @rkomandu, see this comment from June https://github.com/noobaa/noobaa-core/issues/7353#issuecomment-2151581487
